src/execution_plan/ops/shared/update_functions.c

```diff
@@ -180,25 +176,44 @@ void EvalEntityUpdates
 		update = rm_malloc(sizeof(PendingUpdateCtx));
 		update->ge            = entity;
 		update->attributes    = AttributeSet_ShallowClone(*entity->attributes);
-		update->add_labels    = NULL;
-		update->remove_labels = NULL;
 		// add update context to updates dictionary
 		HashTableAdd(updates, (void *)ENTITY_GET_ID(entity), update);
 	} else {
 		// update context already exists
 		update = (PendingUpdateCtx *)HashTableGetVal(entry);
 	}
 
-	if(array_len(ctx->add_labels) > 0 && update->add_labels == NULL) {
-		update->add_labels = array_new(const char *, array_len(ctx->add_labels));
+	if(array_len(ctx->add_labels) > 0 && *add_labels == NULL) {
+		RG_Matrix_new(add_labels, GrB_BOOL, Graph_RequiredMatrixDim(gc->g),
+			Graph_RequiredMatrixDim(gc->g));
```

Comment:
the dimensions should be: number of nodes in the graph X number of labels in the graph